### PR TITLE
Add login and signup

### PR DESCRIPTION
**Pull request by mnicole**
Added login and signup auth flow...
When submitting the forms, it calls fetch and passes the data to the api. When the response is returned, it sets the response in AsyncStorage so it can be read by the app, and also handles redirecting to the right pages.

### CODE REVIEW

File: components/ErrorMessage.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { Content, Text } from 'native-base';
+import { StyleSheet } from 'react-native';
+
+export class ErrorMessage extends React.Component {
```
Comment (mnicole) [inline components/ErrorMessage.js]: generic error message component, we can style it however we want later

File: constants/api.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export const baseUrl = 'http://localhost:5710/';
```
Comment (mnicole) [inline constants/api.js]: we'll have to change this later to check if you're on development or not and switch to the prod url if so

File: screens/AuthLoadingScreen.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    }
+
+    _bootstrapAsync = async () => {
+        const userToken = await AsyncStorage.getItem('userToken');
+        this.props.navigation.navigate(userToken ? 'Main' : 'Auth');
+    };
+
+    render() {
+        return (
+            <View>
```
Comment (mnicole) [inline screens/AuthLoadingScreen.js]: i took this from the react native docs, we'll have to style it later when we figure out how we want it to look.

File: screens/LoginScreen.js [hunk 4]
```diff
@@ -14,34 +14,97 @@ import {
...
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(data)
+        })
+            .then((response) => {
+                if (response.ok) {
+                    return response.json();
+                } else {
+                    throw Error(
+                        'There was an error, please try again.'
```
Comment (mnicole) [inline screens/LoginScreen.js]: had trouble getting at the error message from the API, so just set the message here for now. fetch can't read the response unless you call response.json() on it.

File: screens/SettingsScreen.js [hunk 5]
```diff
@@ -36,7 +48,9 @@ export default class SettingsScreen extends React.Component {
                             <Right />
                         </Header>
                         <Content>
-                            <Text>Settings here...</Text>
+                            <Button full onPress={this._logout}>
```
Comment (mnicole) [inline screens/SettingsScreen.js]: i just put logout here for now, so you can test the full flow (signing up, then logging out by clicking here, then logging in)

**Comment by ernewtoner** [comment]
Looks good :+1: , I can look more into the fetch / response.json() issue and try to get the error message from the API successfully

**Pull request merged**